src/gui/csvImport/CsvImportWidget.cpp

```diff
@@ -131,15 +132,27 @@ void CsvImportWidget::updateTableview() {
 
 void CsvImportWidget::updatePreview() {
 
+    int minSkip = 0;
+    if (m_ui->checkBoxFieldNames->isChecked())
+        minSkip = 1;
     m_ui->labelSizeRowsCols->setText(m_parserModel->getFileInfo());
-    m_ui->spinBoxSkip->setValue(0);
-    m_ui->spinBoxSkip->setMaximum(qMax(0, m_parserModel->rowCount() - 1));
+    m_ui->spinBoxSkip->setValue(minSkip);
+    m_ui->spinBoxSkip->setMaximum(qMax(minSkip, m_parserModel->rowCount() - 1));
 
-    int i;
+    int i, emptyId = 0;
+    QString s;
```

Comment:
Get rid of this variable and just use fieldName. fieldName should also probably be renamed columnName.
This should also go back into the for loop since you only reference it in that scope.